tasks/overwrites.yml

```diff
@@ -33,3 +33,8 @@
   set_fact:
     apache_ports_configuration_items: "{{ __apache_ports_configuration_items }}"
   when: apache_ports_configuration_items is not defined
+
+- name: Define apache_vhosts
+  set_fact:
+    apache_vhosts: "{{ __apache_vhosts }}"
```

Comment:
was trying to use `apache_vhosts: "{{ __apache_vhosts | combine(apache_vhosts,recursive=true) }}"` to be able to merge default dictionary with variables from `default.yml`. However I kept getting the following error:
```
TASK [apache : Add apache vhosts configuration] *********************************************************************************************************************************************
fatal: [owncloud1]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'str object' has no attribute 'servername'"}
```
@xoxys ideas?

Reply:
These variables are list of dicts not dicts :) Besides that, I would suggest to not merge lists. If a user decides to set `apache_vhosts`  this will overwrite the OS defaults as always. If you want to provide the possibility to keep the OS defaults and just extend it with custom vhosts it's common to use `_extra` variables:

```Yaml
# defaults/main.yml

ansible_vhosts_extra: []

# tasks/overwrite.yml

- name: Define apache_vhosts
  set_fact:
    apache_vhosts: "{{ (__apache_vhosts if apache_vhosts is not defined else apache_vhosts) + apache_vhosts_extra  }}"
```

This will append `apache_vhosts_extra` to the OS default `__apache_vhosts`  or `apache_vhosts`  if defined.

Reply:
Thanks for the note, do you think it is necessary? I think we can go without this and go ahead with the merge.

Reply:
No, we could keep the current behavior for now.